internal/consensus/state.go

```diff
@@ -1188,16 +1211,20 @@ func (cs *State) enterPropose(height int64, round int32) {
 
 	// if not a validator, we're done
 	if !cs.Validators.HasAddress(addr) {
-		logger.Debug("propose step; not proposing since node is not in the validator set",
-			"addr", addr,
-			"vals", cs.Validators)
+		if logger.DebugOn() {
+			logger.Debug("propose step; not proposing since node is not in the validator set",
+				"addr", addr,
+				"vals", cs.Validators)
```

Comment:
This is a case that is worth considering, as `cs.Validators` is a complex struct with a string method etc. etc.